NetflixClone/Classes/Scenes/Search/SearchViewController.swift

```diff
@@ -6,43 +6,178 @@
 //
 
 import UIKit
-
+import Combine
 class SearchViewController: UIViewController {
-
-    @IBOutlet weak var searchCollectionView: UICollectionView!
+    // MARK: Properties
+    
+    private var viewModel: SearchViewModel
+    typealias Section = SearchViewModel.Section
+    private var subscribers: [AnyCancellable] = []
+    
+    private enum Defaults {
+        static let itemsPerPage = 10
+        static let imageHeight = 120
+    }
+    
+    private var currentPage = 1
+    private var query: [String] = []
+    // MARK: Views
+    var stackView: UIStackView! = nil
+    var searchBar: UISearchBar! = nil
+    var collectionView: UICollectionView! = nil
+    // MARK: Data sources
+    var dataSource: UICollectionViewDiffableDataSource<Section, MovieSearchMulti>! = nil
+    
+    init(viewModel: SearchViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        searchCollectionView.dataSource = self
-        searchCollectionView.delegate = self
+        configureHierarchy()
+        configureDataSource()
         
-        let nibName = "\(SearchCollectionViewCell.self)"
-        searchCollectionView.register(UINib(nibName: nibName, bundle: nil), forCellWithReuseIdentifier: nibName)
+        viewModel.$result
+            .receive(on: DispatchQueue.main)
+            .sink { result in
+                if let result = result {
+                    self.updateDataSource(movies: result.newMovies)
+                }
+            }
+            .store(in: &subscribers)
     }
-
 }
-
-extension SearchViewController: UICollectionViewDataSource, UICollectionViewDelegate {
-    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return 10
+// MARK: - Configurations
+extension SearchViewController {
+    private func configureHierarchy() {
+        stackView = UIStackView(frame: view.bounds)
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        searchBar = UISearchBar(frame: CGRect(x: 0.0, y: 0.0, width: view.frame.width, height: 48.0))
+        searchBar.delegate = self
```

Comment:
Let's focus in making the layout in XIB files to avoid adding much code in view controller.